src/bot.rs

```diff
@@ -59,151 +60,283 @@ impl Bot {
         }
     }
 
-    /// Creates a new request and adds a JSON message to it. The returned Future contains a the
-    /// reply as a string.  This method should be used if no file is added because a JSON msg is
-    /// always compacter than a formdata one.
-    pub fn fetch_json<'a>(
+    pub fn fetch_json(
         &self,
-        func: &str,
-        msg: &str,
-    ) -> impl Future<Item = String, Error = Error> + 'a {
-        debug!("Send JSON: {}", msg);
-
-        let mut header = List::new();
-        header.append("Content-Type: application/json").unwrap();
-
-        let mut a = Easy::new();
-        a.http_headers(header).unwrap();
-        a.post_fields_copy(msg.as_bytes()).unwrap();
-        a.post(true).unwrap();
-
-        self._fetch(func, a)
+        func: String,
+        msg: String,
+    ) -> impl Future<Item = String, Error = Error> {
+        fetch_json(self.session.clone(), self.key.clone(), func, msg)
     }
 
-    /// Creates a new request with some byte content (e.g. a file). The method properties have to be
-    /// in the formdata setup and cannot be sent as JSON.
-    pub fn fetch_formdata<'a, T>(
+    pub fn fetch_formdata<T>(
         &self,
-        func: &str,
+        func: String,
```

Comment:
This string is formatted in https://github.com/bytesnake/telebot/pull/15/files#diff-dfe207913bb7c9368ddc50e1ecae3f78R235, a reference is therefore appropriate here

Reply:
Since this string is created in the macros, it exists for the `'static` lifetime. This is now reflected in the code.

Reply:
Yes this is an even better solution